firedrake/projection.py

```diff
@@ -127,8 +128,16 @@ def __init__(self, v, v_out, solver_parameters=None, constant_jacobian=True):
                              v_out.function_space())
         self.v = v
         self.v_out = v_out
-
-        if not self._same_fspace:
+        if bcs:
+            try:
+                bcs = tuple(bcs)
+            except TypeError:
+                bcs = (bcs,)
+            if any(bc.function_space() != v_out.function_space() for bc in bcs):
+                raise ValueError("bcs must be enforced on the space of the result function.")
+        self.bcs = bcs
+
+        if not self._same_fspace or self.bcs:
```

Comment:
(the ..., 0, 0, 1 solution can't be right, as <u, v> != 0 for the test function v one node in from the boundary)

For some reason I was thinking about this in 2015 as well; not the first time it's come up.